Modules/arraymodule.c

```diff
@@ -197,20 +197,22 @@ b_setitem(arrayobject *ap, Py_ssize_t i, PyObject *v)
     /* PyArg_Parse's 'b' formatter is for an unsigned char, therefore
        must use the next size up that is signed ('h') and manually do
        the overflow checking */
-    if (!PyArg_Parse(v, "h;array item must be integer", &x))
+    if (!PyArg_Parse(v, "h:array('b').__setitem__", &x)) {
         return -1;
+    }
     else if (x < -128) {
         PyErr_SetString(PyExc_OverflowError,
-            "signed char is less than minimum");
+                        "array item too small to convert to C char");
```

Comment:
I agree.
I will change all such messages (in my patch) in arraymodule.c to `"Python int too small to convert to C char"`, etc.